articles/openshift/oshift-how-apply-security-best-practices.md

```diff
@@ -0,0 +1,70 @@
+Just some container image scanning tools are as follows:
+  -  OpenSCAP-docker
+  -  Black Duck Hub
+  -  JFrog Xray
+  -  Twistlock
+  -  Image-inspector
+  -  Clair
+
+User should also apply 'Principle of least Access' with containers too. Containers should never be run as root user instead always run as a user with just enough privileges to run the required container work load.
+
+- Secrets - Sensitive information like passwords, SSH Keys, tokens, Certificates etc are some examples of what could be considered to be a secret. As such secrets should never be stored in any place in readable format (Source code, GitHub Repo etc). It is imperative that when secrets are required in OpenShift they are set up in OpenShift as OpenShift Secrets. This allows the secrets to be encrypted within OpenShift and accessed only by the relevant containers at the time of need.
+
+- Limiting Resources - Containers should always be set up with resources limits, in the case of a denial of service attack or just a burst of very high load on a service these resource limits will prevent pods from using all available resources on the node they are running on. You can read more about setting resource quotas and limits [here](https://docs.openshift.com/container-platform/3.11/dev_guide/compute_resources.html)
+
+- Privileged ports - TCP/IP ports below port 1024 are considered to be Privileged Ports and should not be used to front any Service that is deployed into OpenShift.
+
+- Route security - Within OpenShift users can expose their services using routes. Creating a route to a service allows the user to access the service through a URL on a browser. By default, the URLS which are created are exposed on HTTP port 80 without any SSL certificate. Users can secure these routes with SSL encryption using one of three SSL termination methods:
+     1. Edge - SSL termination is done on the HAProxy router from which point onwards the traffic is un-encrypted.
+     2. Passthrough termination - No SSL de-Cryption on the routers instead connections are passed straight through the routers to the OpenShift apps which are then responsible for SSL certificate termination and traffic decryption.
+     3. Re-Encrypt - This process uses 2 certificates, first certificate is held on HAProxy router and the second in the pods. Whereby securing the traffic all the way down to the application.
+
+It is possible to further securing routes by applying IP Whitelisting on routes, we have documented this process here [IP Whitelisting](https://docs.ukcloud.com/articles/openshift/oshift-how-restrict-access-to-openshift-routes-by-ip-address.html).
+
+UKCloud can whitelist source IP's/Subnets in OpenShift to the OpenShift GUI/Metrics page and the services that customers deploy which are exposed on various community networks with the use of security groups. These restrictions can be applied at the time cluster deployment or later through a service request to UKCloud OpenShift team.
```

Comment:
I can't make out what this first sentence is trying to say...

Second sentence:
We can apply these restrictions at the time of cluster deployment or later through a service request via the UKCloud Portal.